mllib/src/main/scala/org/apache/spark/ml/tree/Node.scala

```diff
@@ -269,12 +268,19 @@ private[tree] class LearningNode(
   /**
    * Convert this [[LearningNode]] to a regular [[Node]], and recurse on any children.
    */
-  def toNode: Node = {
-    if (leftChild.nonEmpty) {
-      assert(rightChild.nonEmpty && split.nonEmpty && stats != null,
+  def toNode(prune: Boolean = true): Node = {
```

Comment:
If you just overload the method then you don't need to change the existing function calls.

```scala
def toNode: Node = toNode(prune = true)
```